frontend/src/services/ApiService.tsx

```diff
@@ -0,0 +1,44 @@
+import ServiceProvider from "./ServiceProvider";
+import ProvidedServices from "./ProvidedServices";
+import axios from "axios";
+import { useEffect, useRef, useState } from "react";
+
+export interface IApiService {
+  login(user: string, pass: string): void;
+}
+
+const ApiServiceContext = ServiceProvider.createContext(
+  ProvidedServices.ApiService
+);
+export const useApiService = () =>
+  ServiceProvider.use<IApiService>(ProvidedServices.ApiService);
+
+export function ApiService({ children }: any) {
+  const [token, setToken] = useState("");
```

Comment:
I was talking with Joey about this, I don't think we need to maintain the token as global state at the moment.